sw/device/tests/otbn_ecdsa_op_irq_test.c

```diff
@@ -136,20 +136,7 @@ static void otbn_wait_for_done_irq(dif_otbn_t *otbn) {
 
   // At this point, OTBN should be running. Wait for an interrupt that says
   // it's done.
-  while (true) {
-    // This looks a bit odd, but is needed to avoid a race condition where the
-    // OTBN interrupt comes in after we load the otbn_finished flag but before
-    // we run the WFI instruction. The trick is that WFI returns when an
-    // interrupt comes in even if interrupts are globally disabled, which means
-    // that the WFI can actually sit *inside* the critical section.
-    irq_global_ctrl(false);
-    if (plic_peripheral != UINT32_MAX) {
-      break;
-    }
-    wait_for_interrupt();
-    irq_global_ctrl(true);
-  }
-  irq_global_ctrl(true);
+  ATOMIC_WAIT_FOR_INTERRUPT(plic_peripheral == kTopEarlgreyPlicPeripheralOtbn);
```

Comment:
I would check for `plic_peripheral != UINT32_MAX` to avoid changing the test too much.

Reply:
done